code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -54,6 +56,12 @@
 			return
 	return
 
+/obj/item/weapon/storage/proc/empty_contents_to(var/atom/place)
+	var/turf = get_turf(place)
+	for(var/obj/objects in contents)
+		remove_from_storage(objects, turf)
+		objects.pixel_x = rand(-6,6)
+		objects.pixel_y = rand(-6,6)

```

Comment:
You can contextually click things onto tables to get a pixel offset depending on where you place them, so something else is correcting for that so it should not be an issue.